README.md

```diff
@@ -15,6 +47,507 @@ go get -u github.com/src-d/kallax/...
+* Slices or arrays of structs (or pointers to structs) that are models themselves will be considered a 1:N relationship.
+* A struct or pointer to struct field that is a model itself will be considered a 1:1 relationship.
+* For relationships, the foreign key is assumed to be the name of the model converted to lower snake case plus `_id` (e.g. `User` => `user_id`). You can override this with the struct tag `fk:"my_custom_fk"`.
+* For inverse relationship, you need to use the struct tag `fk:",inverse"`. You can combine the `inverse` with overriding the foreign key with `fk:"my_custom_fk,inverse"`. In the case of inverses, the foreign key name does not specify the name of the column in the relationship table, but the name of the column in the own table. The name of the column in the other table is always the primary key of the other model and cannot be changed for the time being.
+* Foreign keys *do not have to be in the model*, they are automagically managed underneath by kallax.
+
+Kallax also provides a `kallax.Timestamps` struct that contains `CreatedAt` and `UpdatedAt` that will be managed automatically.
+
+Let's see an example of models with all these cases:
+
+```go
+type User struct {
+        kallax.Model       `table:"users"`
+        kallax.Timestamps
+        kallax.ID int64    `pk:"autoincr"`
+        Username  string
+        Password  string
+        Emails    []string
+        // This is for demo purposes, please don't do this
+        // 1:N relationships load all N rows by default, so
+        // only do it when N is small.
+        // If N is big, you should probably be querying the posts
+        // table instead.
+        Posts []*Post `fk:"poster_id"`
+}
+
+type Post struct {
+        kallax.Model      `table:"posts"`
+        kallax.Timestamps
+        ID       int64    `pk:"autoincr"`
+        Content  string   `kallax:"post_content"`
+        Poster   *User    `fk:"poster_id,inverse"`
+        Metadata Metadata `kallax:",inline"`
+}
+
+type Metadata struct {
+        MetadataType MetadataType
+        Metadata map[string]interface{} // this will be json
+}
+```
+
+### Struct tags
+
+| Tag | Description | Can be used in |
+| --- | --- | --- | --- |
+| `table"table_name"` | Specifies the name of the table for a model | embedded `kallax.Model` |
+| `pk:""` | Specifies the field is a primary key | any field with a valid identifier type |
+| `pk:"autoincr"` | Specifies the field is an auto-incrementable primary key | any field with a valid identifier type |
+| `kallax:"column_name"` | Specifies the name of the column | Any model field that is not a relationship |
+| `kallax:"-"` | Ignores the field and does not store it | Any model field |
+| `kallax:",inline"` | Adds the fields of the struct field to the model. Column name can also be given before the comma | Any struct field |
```

Comment:
now it is said: `Column name can also be given before the comma`, what is partially true;
it should be confirmed here in the readme, that if the inline contains more than one property, the column name before the `inline` tag will be ignored; otherwise the user could think that the provided column_name will be used for the column name generation in any way (as a prefix... or any other possible way)